### PR TITLE
Shadow DOM: prune bundle size impact

### PR DESCRIPTION
**Pull request by spmonahan**
## Previous Behavior

The Shadow DOM feature in Fluent v8 introduces a large bundle size hit for users who are not using the shadow DOM feature.

## New Behavior

There is still a bundle size hit but it has be reduced a fair bit. There is more work to do here I think but this is a good start and ready to be merged into `shadow-dom`.

This refactor focuses on two things:
1. Minimizing the bundle size increase due to `@fluentui/merge-styles`. This is accomplished by refactoring `Stylesheet` to have less code and moving optional functions out of `Stylesheet` so they become pay-to-play.
2. Refactoring the Shadow DOM React contexts to become dependency injection containers. In this refactored implementation the various shadow DOM features become no-ops (and therefore much smaller in terms of bundle size) but we can still easily inject the actual shadow DOM behavior for users who opt in.

### CODE REVIEW

**Comment by fabricteam** [comment]
[w] TypeError: Cannot destructure property 'commitSHA' of 'result[(result.length - 1)]' as it is undefined.
[w] Failed to fetch report from the remote. Retrying...
[w] TypeError: Cannot destructure property 'commitSHA' of 'result[(result.length - 1)]' as it is undefined.
[w] Failed to fetch report from the remote. Retrying...
[w] TypeError: Cannot destructure property 'commitSHA' of 'result[(result.length - 1)]' as it is undefined.
[w] Failed to fetch report from the remote. Retrying...
[w] TypeError: Cannot destructure property 'commitSHA' of 'result[(result.length - 1)]' as it is undefined.
[w] Failed to fetch report from the remote. Retrying...
[w] TypeError: Cannot destructure property 'commitSHA' of 'result[(result.length - 1)]' as it is undefined.
[w] Failed to fetch report from the remote. Retrying...
<div id="bundleSizeComment9423"/>

File: packages/style-utilities/src/index.ts [hunk 1]
```diff
@@ -1,9 +1,110 @@
-export * from './classNames/index';
-export * from './styles/index';
-export * from './utilities/index';
-export * from './interfaces/index';
-export * from './MergeStyles';
-export * from './cdn';
+export { AnimationClassNames, FontClassNames, ColorClassNames } from './classNames/index';
```
Comment (spmonahan) [inline packages/style-utilities/src/index.ts:1]: This exports the same API. The change just addresses a lint warning to be specific about exports (i.e., no `export * ...`)

File: packages/utilities/src/shadowDom/contexts/MergeStylesShadowRootContext.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+export type MergeStylesShadowRootContextValue = {
+  /**
+   * Map of stylesheets available in the context.
+   */
+  stylesheets: Map<string, CSSStyleSheet>;
+
+  /**
+   * Shadow root for this context.
+   */
+  shadowRoot?: ShadowRoot | null;
```
Comment (khmakoto) [inline packages/utilities/src/shadowDom/contexts/MergeStylesShadowRootContext.tsx:14]: Is the `ShadowRoot` type implicit here? I don't see an import with it...

Comment (spmonahan) [reply packages/utilities/src/shadowDom/contexts/MergeStylesShadowRootContext.tsx:14]: [`ShadowRoot` is from TS's DOM lib types](https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L21376).

File: packages/merge-styles/src/Stylesheet.ts [hunk 3]
```diff
@@ -238,114 +251,31 @@ export class Stylesheet {
...
-    this.forEachAdoptedStyleSheet((srcSheet, key) => {
-      const clonedSheet = cloneExtendedCSSStyleSheet(srcSheet, this.makeCSSStyleSheet(targetWindow));
-      targetStylesheet.addAdoptableStyleSheet(key, clonedSheet);
-    });
-
-    if ((global as Window)?.document) {
-      const globalStyles = (global as Window).document.querySelectorAll('[data-merge-styles-global]') || [];
-      for (let i = 0; i < globalStyles.length; i++) {
-        const styleTag = targetWindow.document.createElement('style');
-        // TODO: insert this in the right place
```
[the remark this answers is not in the document]
Comment (spmonahan) [reply packages/merge-styles/src/Stylesheet.ts:337]: No, I need to grab that.

**Pull request merged**